src/test/java/pl/widulinski/webcrawlertool/ScrAppApplicationTests.java

```diff
@@ -0,0 +1,205 @@
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    String pattern = "dd.MM.yyyy";
+    String dateInStringTest = new SimpleDateFormat(pattern).format(new Date());
+    Set<FoundWebElement> webElementsList = new HashSet<>();
+    FoundWebElement webElementTest = new FoundWebElement();
+    PreparedDataToSrchDto preparedDataToSearchTest = preparedDataToSrchDto;
+    Map<String, DataToScrap> elements = new HashMap<>();
+
+
+    @Test
+    @Order(2)
+    public void urlControllerTest() throws Exception {
+        mockMvc
+                .perform(get("/"))
+                .andExpect(status().isOk())
+                .andExpect(view().name("index"))
+                .andExpect(content().string(containsString("Select Shop")));
+
+        mockMvc
+                .perform(get("/adminMenu"))
+                .andExpect(status().isOk())
+                .andExpect(view().name("adminMenu"))
+                .andExpect(content().string(containsString("WebShop Name")));
+    }
+
+
+
+    @Test
+    @Order(3)
+    public void webDriverTest() {
+
+        System.setProperty("webdriver.chrome.driver", "C:\\chromedriver.exe");
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("headless");
+        WebDriver driver = new ChromeDriver(options);
+        String baseUrl = "https://www.allegro.pl";
+        driver.get(baseUrl);
+
+        driver.manage().timeouts().implicitlyWait(3, TimeUnit.SECONDS);
+
+        if (driver.findElement(By.xpath("//button[text()='przejdź dalej' and @class='_13q9y _8hkto _11eg6 _7qjq4 _ey68j']")).isEnabled()) {
+            driver.findElement(By.xpath("//button[text()='przejdź dalej' and @class='_13q9y _8hkto _11eg6 _7qjq4 _ey68j']")).click();
+        }
+
+        assertEquals("Allegro - atrakcyjne ceny i 100% bezpieczeństwa", driver.getTitle());  //testing does opened page using webdriver
+
+
+        driver.close();
+    }
+
+
+    @Test
+    @Order(4)
+    public void getDataFromRepositoryTest() {
+
+
+        for (DataToScrap element : dataToScrapRepository.findAll()) {
+
+            elements.put(element.getShop() + "_" + element.getCategory(), element);
+        }
+        assertFalse(elements.isEmpty());
+    }
+
+
+    @Test
+    @Order(5)
+    public void getLastPageTest() {
+
+        System.setProperty("webdriver.chrome.driver", "C:\\chromedriver.exe");
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("headless");
+        WebDriver driver = new ChromeDriver(options);
+
+        elements.forEach((key, value) -> {
+
+            driver.get(value.getUrlToCategory() + "1");
+
+            driver.manage().timeouts().implicitlyWait(3, TimeUnit.SECONDS);
+
+            WebElement webElement = driver.findElement(By.xpath(value.getXpathOfLastPage().replace("|", "'")));
+
+            assertTrue(webElement.isEnabled());
+
+        });
+
+        driver.close();
+    }
+
+    @Test
+    @Order(6)
+    public void scrapArticleDataTest() {
```

Comment:
Tu podobnie co test z 87 - jedyna roznica ze dodałes tu wywołanie swojego repo - nic to nie zmienia, nadal testujesz Drivera a nie swoj kod

Reply:
test zmodywikowany - testuje dane zeskrobane i wrzucone do pliku xlsx